src/modules/EditUserProfile/components/EditUserProfileUI/index.tsx

```diff
@@ -1,47 +1,100 @@
-import './edit-user-profile.scss';
-
 import React, {
-  ReactElement,
+  type MutableRefObject,
   useRef,
   useState,
   useContext,
 } from 'react';
-import { useEditUserProfileContext } from '../../context/EditUserProfileProvider';
 
-import Modal from '../../../../ui/Modal';
-import { LocalizationContext } from '../../../../lib/LocalizationContext';
+import { User } from '@sendbird/chat';
+import './edit-user-profile.scss';
+
 import useSendbirdStateContext from '../../../../hooks/useSendbirdStateContext';
+import { useEditUserProfileContext } from '../../context/EditUserProfileProvider';
+import { LocalizationContext } from '../../../../lib/LocalizationContext';
+import { SendBirdState } from '../../../../lib/types';
+import { USER_ACTIONS } from '../../../../lib/dux/user/actionTypes';
 
-import Input, { InputLabel } from '../../../../ui/Input';
-import Avatar from '../../../../ui/Avatar';
-import Icon, { IconTypes } from '../../../../ui/Icon';
+import Modal from '../../../../ui/Modal';
 import { ButtonTypes } from '../../../../ui/Button';
-import Label, { LabelColors, LabelTypography } from '../../../../ui/Label';
-import TextButton from '../../../../ui/TextButton';
-import { noop } from '../../../../utils/utils';
-import { USER_ACTIONS } from '../../../../lib/dux/user/actionTypes';
+import { EditUserProfileUIView } from './EditUserProfileUIView';
 
-export default function EditUserProfileUI(): ReactElement {
-  const editProfileProps = useEditUserProfileContext();
-  const store = useSendbirdStateContext();
-  const hiddenInputRef = useRef(null);
+interface HandleUpdateUserInfoParams {
+  globalContext: SendBirdState;
+  formRef: MutableRefObject<any>;
+  inputRef: MutableRefObject<any>;
+  profileImage: File;
+  onEditProfile?: (user: User) => void;
+}
+const handleUpdateUserInfo = ({
+  globalContext,
+  formRef,
+  inputRef,
+  profileImage,
+  onEditProfile,
+}: HandleUpdateUserInfoParams) => {
+  const { stores, dispatchers } = globalContext;
+  const sdk = stores.sdkStore.sdk;
+  const user = stores.userStore.user;
+  const { userDispatcher } = dispatchers;
+
+  if (user?.nickname !== '' && !inputRef.current.value) {
```

Comment:
why we need to check empty nickname here?

Reply:
Because it's meanless to update the user nickname with empty string, so we return nothing when it's empty

Reply:
so its || not && ?

Reply:
The `user.nickname` is for the nickname
and the `inputRef.current.value` is for the profile image

So we can ignore updating when both inputs are not given